src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/tests/test_backup_commands.py

```diff
@@ -0,0 +1,526 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from datetime import datetime, timedelta
+
+from azure.cli.testsdk import ScenarioTest, JMESPathCheck, JMESPathCheckGreaterThan, JMESPathCheckExists, ResourceGroupPreparer, StorageAccountPreparer
+from .preparers import VaultPreparer, VMPreparer, ItemPreparer, PolicyPreparer, RPPreparer
+
+from azure.mgmt.recoveryservices.models import StorageModelType
+
+class BackupTests(ScenarioTest):
+    def _get_first_if_list(self, json_response):
+        json_obj = None
+        if isinstance(json_response, list):
+            if len(json_response) > 0:
+                json_obj = json_response[0]
+        else:
+            json_obj = json_response
+        return json_obj
+
+    def _get_vm_version(self, vm_type):
+        if vm_type == 'Microsoft.Compute/virtualMachines':
+            return 'Compute'
+        elif vm_type == 'Microsoft.ClassicCompute/virtualMachines':
+            return 'Classic'
+
+    @ResourceGroupPreparer()
+    @VaultPreparer()
+    @VMPreparer()
+    @StorageAccountPreparer()
+    def test_backup_restore(self, resource_group, resource_group_location, vault_name, vm_name, storage_account):
+        # Test Flags
+        do_enable_protection = True
+        do_get_container = True
+        do_get_item = True
+        do_trigger_backup = True
+        do_get_recovery_point = True
+        do_trigger_restore = True
+        do_disable_protection = True
```

Comment:
I don't understand the purpose of these test flags. They're not changed in anyway, not participate in data driven test.